### ISSUE TITLE
<ignoredFolder> config option has no effect

### ISSUE THREAD

**Issue by mythsmith** [syncthing/syncthing#6820]
Version 1.7.0

OS Ubuntu 18.04

ignoredFolder, child of the main <configuration> node, should have this effect:

```
    Contains the ID of the folder that should be ignored. This folder will
    always be skipped when advertised from a remote device, i.e. this will be
    logged, but there will be no dialog about it in the web GUI.
```

https://github.com/syncthing/docs/blame/main/users/config.rst#L169

I could not have it working neither by setting to a remote folder being shared with me as `<ignoredFolder>folder-id</ignoredFolder>` under `<configuration>` entry, nor by reproducing what syncthing does and setting `<ignoredFolder time="some-time-stamp" id="folder-id" label="folder-label></ignoredFolder>` under the `<device>` entry sharing that folder.

The effect is invariably that the xml entry gets removed from the config, and a new `<pendingFolder time="another-time-stamp" id="folder-id" label='folder-label"></ignoredFolder>` gets created under the `<device>` sharing that folder. This causes a warning to appear in the UI, which persists until acknowledged. 
If acknowledged, a new entry `<ignoredFolder time="still-another-time-stamp" id="folder-id" label="folder-label"></ignoredFolder>` is added to the interested `<device>` entry, which is persistent upon restart.

To summarize:

- `<configuration>...<ignoredFolder>` has no effect and gets removed upon syncthing startup.
- `<device>...<ignoredFolder time...>` has no effect and gets replaced by `<pendingFolder time...>` upon syncthing startup
- if ignore is acknowledged, syncthing creates a `<device>...<ignoredFolder time...>` entry which is apparently identical to the one set manually, but this one works and is not replaced by a `<pendingFolder>` entry.

Apparently I am out of options to ignore a folder by just editing the config.xml.

**Comment by AudriusButkevicius** [comment]
I've justed tested it and it works from the UI.
![image](https://user-images.githubusercontent.com/1144861/86938127-f837f400-c137-11ea-9e3c-d289476e8269.png)

Please open a documentation issue in the documentation repo.

As for why it doesn't work for you, my gut says that you are modifying the xml file on a running instance and expect to see changes happen.

This is not how it works, syncthing reads the config file on startup and then pretty much never again, so making changes to the file while it's running has no effect, and any changes made to the file are guaranteed to be overwritten whenever it decides to save it's in memory copy of the config.

If you have any further questions, please use the forum for support.

**Comment by mythsmith** [comment]
I was obviously testing with syncthing shut down, modify xml file, start syncthing.

I see that config_test.go never verifies the case where the `<ignoredFolder>folder-id</ignoredFolder>` is added to the root `<configuration>` node:
https://github.com/syncthing/syncthing/blob/fdf5a5c0d78413ba489c05b3fd08f3ed72493a3d/lib/config/config_test.go#L812

Because this eventuality is not included in the `ignoredfolders.xml` test data:
https://github.com/syncthing/syncthing/blob/fdf5a5c0d78413ba489c05b3fd08f3ed72493a3d/lib/config/testdata/ignoredfolders.xml#L4

Instead, ignores are inserted under each `<device>` node, as  `<device>...<ignoredFolder id="folder-id"></ignoredFolder>` (so without any `time` and `label` attributes set).

I will leave to the testers to find out which other conditions break the IgnoredFolder mechanics.

**Comment by AudriusButkevicius** [comment]
I believe it used to be under configuration, then we changed to be under devices, and performed a config migration. The docs and the tests you are referring to refer to the old format which is no longer valid.

The tests pass because it runs the migration when it loads the old config text file.

**Comment by mythsmith** [comment]
Anyhow, I tested using the same structure as `ignoredfolders.xml` and that way it works perfectly, thanks!
